src/table.rs

```diff
@@ -328,13 +328,18 @@ impl<'p, T: Slot> PageView<'p, T> {
 impl Page {
     #[inline]
     fn new<T: Slot>(ingredient: IngredientIndex, memo_types: Arc<MemoTableTypes>) -> Self {
-        #[cfg(not(loom))]
+        #[cfg(not(feature = "shuttle"))]
         let data: Box<PageData<T>> =
             Box::new([const { UnsafeCell::new(MaybeUninit::uninit()) }; PAGE_LEN]);
 
-        #[cfg(loom)]
-        let data: Box<PageData<T>> =
-            Box::new([const { MaybeUninit::uninit() }; PAGE_LEN].map(UnsafeCell::new));
+        #[cfg(feature = "shuttle")]
+        let data = unsafe {
+            let data = (0..PAGE_LEN)
+                .map(|_| UnsafeCell::new(MaybeUninit::uninit()))
+                .collect::<Box<[PageDataEntry<T>]>>();
```

Comment:
We could consider using this branch in debug builds because Rust doesn't inline the call without optimizations enabled. We've seen this in ty where the `Project` table takes 0.5 of stack frame!